src/libs/actions/Subscription.ts

```diff
@@ -279,6 +292,44 @@ function clearOutstandingBalance() {
     API.write(WRITE_COMMANDS.CLEAR_OUTSTANDING_BALANCE, null, onyxData);
 }
 
+function cancelBillingSubscription(cancellationReason: FeedbackSurveyOptionID, cancellationNote: string) {
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.NVP_PRIVATE_CANCELLATION_DETAILS,
+            value: [
+                ...cancellationDetails,
+                {
+                    errors: undefined,
+                },
+            ],
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.NVP_PRIVATE_CANCELLATION_DETAILS,
+            value: [
+                ...cancellationDetails,
+                {
+                    cancellationType: undefined,
```

Comment:
We don't want to show any error?

Reply:
What do you mean? If an error is returned from the API it will be saved to Onyx, regarding an error message that could be shown to the user, it hasn't been included in the design doc, but maybe it should have been 🤔